### ISSUE TITLE
Add contract deployment transaction

### ISSUE THREAD

**Issue by ureeves** [dusk-network/phoenix#163]
#### Summary
A transaction type should be added to deal with the deployment of contracts. This is necessary for a [contract deployment system](https://github.com/dusk-network/rusk/issues/1071) to be able to function in the context of phoenix.

The transaction type should allow for charging the transaction, as well as deploying a contract - meaning it should include all the relevant data and metadata to perform this. 

#### Possible solution design or implementation
A possible way to tackle this is to replace the `Transaction::call` field by a new field - maybe called `exec` - that contains an enum that can either be a contract call, or a contract deployment.

```rust
struct Transaction {
    /* other fields */
    exec: Execution,
}

enum Execution {
    Call { /*fields*/ }
    Deploy { /*fields*/ }
}
```

#### Additional context
Upstream should be able to use the data of this transaction to charge the user, as well as make decisions on *how much* to charge. This means the fields of the new transaction should likely be public, or otherwise accessible.

**Comment by moCello** [comment]
Closed as no longer relevant after #175
